source/communication_interfaces/include/communication_interfaces/sockets/UDPSocket.hpp

```diff
@@ -7,6 +7,27 @@
 namespace communication_interfaces::sockets {
 
 /**
+ * @struct UDPSocketConfiguration
+ * @brief Configuration parameters for a UDP sockets
+ */
+struct UDPSocketConfiguration {
+  UDPSocketConfiguration(
```

Comment:
Just sharing an alternative with different pro/cons (see it building it [here](https://onlinegdb.com/kx2UFxkcU)):

```c++
// Definition
struct UDPSocketConfiguration {
  UDPSocketConfiguration() = delete;

  std::string ip_address;
  int port;
  int buffer_size;
  bool enable_reuse = false;
  double timeout_duration_sec = 0.0;
};

...

// Usage

// ERROR: do not compile
createUDPSocket(
  UDPSocketConfiguration(),
);

// OK: works, can add as many field as you want, if not specified a field will take its default value, if they don't have one they are zero'd 
createUDPSocket(
  UDPSocketConfiguration{
    .ip_address = "127.0.0.1",
    .port = 4242,
    .buffer_size = 1337,
  }
);

// OK: biggest con, no easy way to force mandatory fields to be specified
createUDPSocket(
  UDPSocketConfiguration{
    .ip_address = "127.0.0.1",
  }
);
```

Pros:
 - Easier to see default values
 - Easier to add new fields
 - Less boilerplate
 - Clearer to see what fields as which value at initialization point
 - (TBC) If you only add new fields with default without changing any other types, it might not change the ABI, meaning you don't need to recompile users of this struct (less breaking changes -> more backward compatibility)

Cons:
 - No way to check if mandatory fields are set at compile-time, need run-time checks

Reply:
Okay, looks nice as well. I tried it for the UDP config in the most recent commit. If you guys like it better that way, I'll do the same for the ZMQ config

Reply:
The problem for the ZMQ config is that we have the field `bind_socket` which is a bool with no default value. If the user omits it, there is no warning or anything and it might just be the reason why the app doesn't work...